pkg/controller/deployment/deployment_controller.go

```diff
@@ -342,15 +381,26 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
-	if d.DeletionTimestamp != nil {
-		return dc.syncStatusOnly(d)
+	deployments, err := dc.dLister.Deployments(d.Namespace).List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error listing deployments in namespace %s: %v", d.Namespace, err)
 	}
 
 	// Handle overlapping deployments by deterministically avoid syncing deployments that fight over ReplicaSets.
-	if err = dc.handleOverlap(d); err != nil {
+	overlaps, err := dc.handleOverlap(d, deployments)
+	if overlaps {
```

Comment:
`handleOverlap` now can return errors even if the current deployment does not overlap eg. it was marked as overlapping before but not anymore but the controller failed to remove the overlapping annotation.